### PR TITLE
Add Sign In, Sign Up, and Forgot/Reset Password Pages

### PR DESCRIPTION
**Pull request by Walsker**

### CODE REVIEW

File: src/shared/theme.tsx [hunk 1]
```diff
@@ -1,12 +1,14 @@
...
-    return 'light';
-  }
+  // const time = new Date().getHours();
+  // if (time > 18 || time < 7) {
+  //   return 'dark';
+  // } else {
+  //   return 'light';
+  // }
+
+  return 'dark';
```
Comment (dellisd) [inline src/shared/theme.tsx:11]: Is the login form designed only for the dark theme?

**Review by dellisd — COMMENTED** [review]
The only general comment I have is that the error handling is basically non-existent.
A simple error handling scheme would be to just use a string union type to represent each possible failure case and then turn that into a message in the user-facing component.

File: src/shared/theme.tsx (hunk 1 above)
Comment (Walsker) [reply src/shared/theme.tsx:11]: Yeah... I didn't want to import this theme variable and have if statements everywhere, so I rendered it useless for the time being

**Comment by Walsker** [comment]
Most errors get _handled_ but not actually displayed to the user. I'm okay with merging this and working on the rest in the morning.

File: src/shared/theme.tsx (hunk 1 above)
Comment (dellisd) [reply src/shared/theme.tsx:11]: I don't know why we didn't just use a media query to begin with, but I guess that'll be for a separate PR.

**Review by dellisd — APPROVED** [review]

**Pull request merged**